egui_web/src/webgl1.rs

```diff
@@ -485,10 +495,175 @@ impl crate::Painter for WebGlPainter {
                 ));
             }
         }
+
+        self.post_process.end();
+
         Ok(())
     }
 }
 
+/// Uses a framebuffer to render everything in linear color space and convert it back to sRGB
+/// in a separate "post processing" step
+struct PostProcess {
+    gl: Gl,
+    pos_buffer: WebGlBuffer,
+    a_pos_loc: u32,
+    index_buffer: WebGlBuffer,
+    texture: WebGlTexture,
+    texture_size: (i32, i32),
+    fbo: WebGlFramebuffer,
+    program: WebGlProgram,
+}
+
+impl PostProcess {
+    fn new(gl: Gl, width: i32, height: i32) -> Result<PostProcess, JsValue> {
+        let fbo = gl
+            .create_framebuffer()
+            .ok_or("failed to create framebuffer")?;
+        gl.bind_framebuffer(Gl::FRAMEBUFFER, Some(&fbo));
+
+        let texture = gl.create_texture().unwrap();
+        gl.bind_texture(Gl::TEXTURE_2D, Some(&texture));
+        gl.tex_parameteri(Gl::TEXTURE_2D, Gl::TEXTURE_WRAP_S, Gl::CLAMP_TO_EDGE as i32);
+        gl.tex_parameteri(Gl::TEXTURE_2D, Gl::TEXTURE_WRAP_T, Gl::CLAMP_TO_EDGE as i32);
+        gl.tex_parameteri(Gl::TEXTURE_2D, Gl::TEXTURE_MIN_FILTER, Gl::NEAREST as i32);
+        gl.tex_parameteri(Gl::TEXTURE_2D, Gl::TEXTURE_MAG_FILTER, Gl::NEAREST as i32);
+        gl.pixel_storei(Gl::UNPACK_ALIGNMENT, 1);
+        // TODO: https://developer.mozilla.org/en-US/docs/Web/API/EXT_sRGB
+        // Dark colors are slightly wrong when not using SRGB8_ALPHA8 format
```

Comment:
This will mean any images shown with the WebGL1 backend (e.g. on iPhones) will have a lot of banding in the dark regions. For instance, in the "HTTP" demo tab you can enter https://picsum.photos/seed/298.393/640/480 and see really ugly banding on an iPhone with this new render-to-texture approach.

It also makes the egui window shadows look really bad in dark mode.

I think we should probably revert the changes to the WebGL1 pipeline, and only do this render-to-texture trick in WebGL2.

Reply:
Makes sense, will revert